### ISSUE TITLE
Clarify Event types pause/stop and resume/start

### ISSUE THREAD

**Issue by edwinvandenbelt** [openmobilityfoundation/mobility-data-specification#758]
### Is your feature request related to a problem? Please describe.

As a city, I would like to get insights into the intermediate destinations of station-based assets. In most cases, station-based assets are returned to the original station.

### Describe the solution you'd like

These movements can only be detected by tracing where the assets are locked (=pauzed) during the trip. 
Add at least 'pauzed' to the vehicle_event:
https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/general-information.md#event-types

### Is this a breaking change

 * I'm not sure

### Impacted Spec

 * `provider`

### Describe alternatives you've considered

Demand in the trips endpoint that locations that are used as stops, must be provided in the `route` field.

**Comment by S-eb** [comment]
@jyeo17

**Comment by schnuerle** [comment]
In Event Types, there are some values that may be equivalent for pausing trips. 

https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/modes/event_types.md

For example `trip_pause`, `trip_resume`, `reservation_stop`, etc. 

How is your idea different that what these provide now?

**Comment by edwinvandenbelt** [comment]
ha ha ha! They are already added on the 25th of April by Kegan. What's the formal process? Should I close this issue?

**Comment by schnuerle** [comment]
I think we can clarify pause/stop and resume/start for each mode in MDS 2.1. Hopefully without it being a breaking change.

**Comment by schnuerle** [comment]
Here are all the relevant **trip** related event types. 

`event_type` | Description
-- | --
trip_cancel | Cancel trip
trip_end | End trip
trip_pause | Pause trip temporarily but do not end trip
trip_resume | Resume trip
trip_start | Start trip
trip_stop | Stop trip

This logic is how I make sense of this:

1. trip_start - first start of a trip and transaction (required for any trip)
2. trip_pause - trip intends to continue, but is paused in motion or transaction
3. trip_resume - trip resumes after a pause. A pause is required before a resume can happen
4. trip_cancel - trip is ended by anyone for some reason, either before or after motion starts
5. trip_stop - last movement in the trip, intending to end it. Customer may be idling or interacting with devices to end trip.
6. trip_end - end of the trip and transaction

How does this look, and is it your understanding of event types? Are these descriptions good and can be added to spec to clarify these event types? Is trip_stop needed?

There are also these events which can affect a trip (my comments in parentheses)...

`event_type` | Description
-- | --
driver_cancellation | Driver cancelled a trip (if known, and more specific than trip_cancel)
passenger_cancellation | Passenger cancelled a trip (if known, and more specific than trip_cancel)
provider_cancellation | Provider cancelled a trip (if known, and more specific than trip_cancel)
remote_start | Remotely start the engine
remote_end | Remotely stop the engine
reservation_cancel | Reservation cancelled before trip (different than trip_cancel since the transaction hadn't started yet, and the reservation hold is let go)
reservation_start | Reservation started
reservation_stop | Reservation stopped temporarily (not sure when this is used. I guess it can go from here to start or cancel)

**Comment by edwinvandenbelt** [comment]
Hi Michael!
Sounds plauseble. In the TOMP-API we more or less use the same events, except the 'stop trip'. It is hard to determine whether it is a start of a pause, or the start of an intended end.
We only use: start, stop, pause, resume.

The cancel is something that is outside the execution, it can be done before the trip is started (in that case, it will never be reported, or you should have something like 'trip_booked'), or when an incident/accident happens, it will require other steps ('trip_interrupted'? Optionally 'trip_redressed'). We have a dedicated module for this.

imho, trip_stop is not relevant. Trip_cancel is not relevant either, or a few other events should be added to cope with support events.

**Comment by schnuerle** [comment]
Alright the next step for this is to create a PR with clarifications on all of the event definitions, and new ones for the incident capture discussion.

**Comment by jiffyclub** [comment]
Some additional fields it would be nice to have that we've talked about at Populus would be the ability to know whether a trip stop is for passenger pick up or drop off. There are places that charge fees to operators based on pickups and drop offs and in the current spec there's nothing to indicate whether a stop is for one of those events and which one it is (in case they charge differently for a pickup vs a drop off).

**Comment by schnuerle** [comment]
@jiffyclub When you say stop, do you meant the MDS Stop (just want to clarify it's not that)? Or just the start or end of a trip where there is a passenger pickup or drop off?

If the latter, are you thinking maybe two new fields in the trip attributes, like `passenger_pickup` and `passenger_drop_off`? As a boolean to flag whether the trip had these? 

https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/modes/passenger-services.md#trip-attributes

Could `passenger_pickup` and `passenger_drop_off` become Events instead? Which way is best?

**Comment by schnuerle** [comment]
See this [commit](https://github.com/openmobilityfoundation/mobility-data-specification/pull/965/commits/2a89af1d442f629541b7b9f689c332b1a4c0859d) for clarification of event type descriptions. Any feedback on this?

**Comment by schnuerle** [comment]
We will be reviewing this at our next public [working group meeting](https://github.com/openmobilityfoundation/mobility-data-specification/wiki/Web-conference-notes,-2025.01.15-(MDS-Working-Group)) tomorrow. Please leave your comments now, and attend the meeting if you can.

**Comment by jiffyclub** [comment]
As a point of clarification, is it the intent of the spec that passenger pickup and drop-off only occur with `trip_start` and `trip_end` events? Imagine a case where three passengers get in a ride share where each will be dropped off at a different location. Is that one journey with three trips, one trip for each drop off? Or is it one trip with multiple stops?

**Comment by schnuerle** [comment]
> As a point of clarification, is it the intent of the spec that passenger pickup and drop-off only occur with `trip_start` and `trip_end` events? Imagine a case where three passengers get in a ride share where each will be dropped off at a different location. Is that one journey with three trips, one trip for each drop off? Or is it one trip with multiple stops?

Yes I believe this should be separate trips in this case. And trips can overlap too. See [Example 2 here](https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/modes/passenger-services.md#journey-id).

**Comment by jiffyclub** [comment]
If we can safely assume that 1 trip_start = 1 passenger pickup and 1 trip_end = 1 passenger drop off (and passenger pick-up/drop-off occur with no other events) then I think that solves our use case.

**Comment by schnuerle** [comment]
See this [commit](https://github.com/openmobilityfoundation/mobility-data-specification/pull/965/commits/53bbf6f376172664f9edf005198a995d0e819e00) for how this is added to the Passenger mode. Feedback welcome in the next week.

**Comment by jiffyclub** [comment]
Looks good, thanks.

**Comment by schnuerle** [comment]
Closed with PR #965.
